src/main/java/gregtech/loaders/MaterialInfoLoader.java

```diff
@@ -14,415 +15,444 @@
-                new MaterialStack(Materials.Titanium, M * 8), // plate
-                new MaterialStack(Materials.Aluminium, M), // single cable
-                new MaterialStack(Materials.Rubber, M * 2))); // plate
-
-        OreDictUnifier.registerOre(MetaTileEntities.HULL[5].getStackForm(), new ItemMaterialInfo(
-                new MaterialStack(Materials.TungstenSteel, M * 8), // plate
-                new MaterialStack(Materials.Platinum, M), // single cable
-                new MaterialStack(Materials.Rubber, M * 2))); // plate
-
-        OreDictUnifier.registerOre(MetaTileEntities.HULL[6].getStackForm(), new ItemMaterialInfo(
-                new MaterialStack(Materials.RhodiumPlatedPalladium, M * 8), // plate
-                new MaterialStack(Materials.NiobiumTitanium, M), // single cable
-                new MaterialStack(Materials.Rubber, M * 2))); // plate
-
-        OreDictUnifier.registerOre(MetaTileEntities.HULL[7].getStackForm(), new ItemMaterialInfo(
-                new MaterialStack(Materials.NaquadahAlloy, M * 8), // plate
-                new MaterialStack(Materials.VanadiumGallium, M), // single cable
-                new MaterialStack(Materials.Rubber, M * 2))); // plate
-
-        OreDictUnifier.registerOre(MetaTileEntities.HULL[8].getStackForm(), new ItemMaterialInfo(
-                new MaterialStack(Materials.Darmstadtium, M * 8), // plate
-                new MaterialStack(Materials.YttriumBariumCuprate, M), // single cable
-                new MaterialStack(Materials.Rubber, M * 2))); // plate
-
-        OreDictUnifier.registerOre(MetaTileEntities.HULL[9].getStackForm(), new ItemMaterialInfo(
-                new MaterialStack(Materials.Neutronium, M * 8), // plate
-                new MaterialStack(Materials.Europium, M), // single cable
-                new MaterialStack(Materials.Rubber, M * 2))); // plate
-
-
-        OreDictUnifier.registerOre(MetaTileEntities.ENERGY_INPUT_HATCH[3].getStackForm(), new ItemMaterialInfo(
-                new MaterialStack(Materials.StainlessSteel, M * 8), // plate
-                new MaterialStack(Materials.Gold, M * 2), // single cable
-                new MaterialStack(Materials.Rubber, M * 4), // plate
-                new MaterialStack(Materials.BlackSteel, M * 2), // fine wire
-                new MaterialStack(Materials.SteelMagnetic, M / 2) // rod
+        OreDictUnifier.registerOre(MetaTileEntities.HULL[GTValues.ULV].getStackForm(), new ItemMaterialInfo(
+                new MaterialStack(WroughtIron, OrePrefix.plate.getMaterialAmount(WroughtIron) * 8), // plate
+                new MaterialStack(RedAlloy, OrePrefix.cableGtSingle.getMaterialAmount(RedAlloy) * 2), // single cable
+                new MaterialStack(Rubber, OrePrefix.plate.getMaterialAmount(Rubber) * 2), // plate (for the cable)
+                new MaterialStack(Wood, OrePrefix.plate.getMaterialAmount(Wood) * 2))); // plate
```

Comment:
As [Quarri6343](https://github.com/Quarri6343) mentioned in above comment there is possible dupe for some materials if machine hull is crafted via Assembly Machine.

Here demonstration on example with ULV Machine Casing.

Recipe for Crafting Table:
![image](https://user-images.githubusercontent.com/3093101/201404077-c349015c-a097-452b-b1ab-8b048c55d4bc.png)

Recipe for Assembly Machine:
![image](https://user-images.githubusercontent.com/3093101/201404180-c789691b-6c8d-4aca-8c01-6e910d47f185.png)

**New** decomposition recipe in Macerator:
![image](https://user-images.githubusercontent.com/3093101/201404422-01008f4a-df47-4ec3-acad-997658821f82.png)

As can be seen from pictures above when recycling ULV Machine Casing Wood is returned. And if ULV Machine Casing was created in Assembly Machine there is dupe of materials.

Which brings us back to recyclation/decomposition problem. As that would be for long discussion let's skip it and look why this problem happed. Before in here were used recipes from Assembly Machine (and only solid parts) not Crafting Table which means no material was duped (as long recipe in Assembly Machine has same or lower price, using same types or less materials).

For that reasoning I think this PR should only change way materials are calculated not from which source.

Reply:
This is probably applicable on many more recipes down the line, so I will stop here with review.